tests/extract/test_extract_job.py

```diff
@@ -104,22 +103,27 @@ def test_django_cannot_encode_datetime_strings(extract_job_settings):
             list(source.query(sql_query, stream=True, row_tuples=True))
 
 
-def test_extract_job(complex_job, extract_job_settings, now):
+@pytest.mark.freeze_time('2020-07-01', ignore=['mo_threads'])
+def test_extract_job(complex_job, extract_job_settings):
```

Comment:
Use `freeze_time`, notice the `ignore=['mo_threads']`, which prevents the `freeze_time` from messing with the clocks in that module (the source of the "deadlock")

* new event created to trigger in 0.3 seconds from current time
* `freeze_time` rolls clock back to July 1
* event "never" triggers